test/ruby_lsp_rails/hover_test.rb

```diff
@@ -238,6 +151,8 @@ class ActiveRecord::Base
 
       def hover_on_source(source, position)
         with_server(source) do |server, uri|
+          sleep(0.1) while RubyLsp::Addon.addons.first.instance_variable_get(:@client).is_a?(NullClient)
```

Comment:
Same approach as we're using in `definition_test`, but we should really extract this to provide a better experience for addon developers.

Reply:
I agree that we should extract it into a helper. And I also think in that helper we should use `addons.find { |addon| addon.name == "Ruby LSP Rails" }` instead. I think `addons.first` will probably break if we someday add another addon to this project's Gemfile as a development dependency?

Reply:
Yeah, we can extract that part out too, e.g. `RubyLsp::Addon.get(...)`.

Reply:
It does feel like something addons can use during testing 👍

Reply:
https://github.com/Shopify/ruby-lsp/pull/2216